miio/gateway.py

```diff
@@ -145,14 +155,22 @@ def discover_devices(self):
         self._devices = []
 
         for x in range(0, len(devices_raw), 5):
+            # Extract discovered information
+            dev_info = SubDeviceInfo
+            dev_info.sid = devices_raw[x]
+            dev_info.type_id = devices_raw[x+1]
+            dev_info.unknown = devices_raw[x+2]
+            dev_info.unknown2 = devices_raw[x+3]
+            dev_info.fw_ver = devices_raw[x+4]
+
             # Construct DeviceType
             try:
-                device_type = DeviceType(devices_raw[x + 1])
+                device_type = DeviceType(dev_info.type_id)
             except ValueError:
                 _LOGGER.warning(
                     "Unknown subdevice type '%i': %s discovered, of Xiaomi gateway with ip: %s",
-                    devices_raw[x + 1],
-                    devices_raw[x],
+                    dev_info.type_id,
+                    dev_info.sid,
```

Comment:
Print out the whole dev_info object out here, dataclasses print out nicely (and maybe someone, sometime, somewhere will see such a warning and figures out what those unknowns are)

Reply:
you are right, just tested and looks very good